### ISSUE TITLE
Fail to parse negative lookbehind with closing angle bracket

### ISSUE THREAD

**Issue by serch** [ammar/regexp_parser#85]
Hi! I think I found a bug: let's say we want to capture all the letters `y` followed by a closing angle bracket `>` that are not preceded by the letter `x`, we could use this regex:

```
irb(main):003:0> re = Regexp.new("(?<!x)y>")
=> /(?<!x)y>/
irb(main):004:0> ["y>", "py>", "xy>"].map { |s| re.match?(s) }
=> [true, true, false]
```

When we try to parse it, it fails:

```
irb(main):003:0> Regexp::Scanner.scan(re)
/Users/serch/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/regexp_parser-2.6.0/lib/regexp_parser/scanner.rb:2521:in `scan': Premature end of pattern at (missing group closing paranthesis) [1] (Regexp::Scanner::PrematureEndError)
```

without the closing angle bracket it succeeds:

```
irb(main):004:0> Regexp::Scanner.scan(Regexp.new("(?<!x)y"))
=> [[:assertion, :nlookbehind, "(?<!", 0, 4], [:literal, :literal, "x", 4, 5], [:group, :close, ")", 5, 6], [:literal, :literal, "y", 6, 7]]
```

**Comment by jaynetics** [comment]
hi!

that's indeed a bug, and a nice one, too!

this parser is used pretty widely, so i'm actually amazed to see such a simple pattern break it without anyone else noticing for so long. 

i've just released v2.6.1 with a fix.

thanks a lot for the report!

**Comment by serch** [comment]
awesome, thank you for such a quick response and fix!
